src/lib/qnio/nio_server.c

```diff
@@ -430,23 +431,49 @@ qns_server_start(char *node, char *port)
                         nioDbg("Accepted connection on descriptor %d "
                                   "(host=%s, port=%s)\n", infd, hbuf, sbuf);
                     }
+
+                    /* Make the incoming socket non-blocking and add it to the
+                     *  list of fds to monitor. */
+                    s = make_socket_non_blocking(infd);
+                    if (s == -1) {
+                        return (-1);
+                    }
+
                     if(cmn_ctx->ssl_ctx)
                     {
                         ssl = SSL_new(cmn_ctx->ssl_ctx);
                         SSL_set_fd(ssl, infd);
                         ret = SSL_accept(ssl);
                         if (ret <= 0) {
-                            nioDbg("ssl accept failed %d", SSL_get_error(ssl, ret));
-                            return -1;
+                            /* Maybe try SSL_accept again */
+                            nioDbg("SSL_accept failed. Will retry");
+                            ret = SSL_get_error(ssl, ret);
+                            if (ret == SSL_ERROR_WANT_READ || ret == SSL_ERROR_WANT_WRITE) 
+                            {
+                                nioDbg("SSL_error is want read or want write %d", ret);
+                                /* put fd in epoll loop to try again */
+                                event.events = EPOLLIN;
```

Comment:
OpenSSL may be waiting for space to send data so ret needs to be checked:
SSL_ERROR_WANT_READ -> EPOLLIN
SSL_ERROR_WANT_WRITE -> EPOLLOUT